test/unit/pybind/data/exp2syn.json

```diff
@@ -0,0 +1 @@
ryExpression": [{"VarName": [{"Name": [{"String": [{"name": "g"}]}]}]}, {"BinaryOperator": [{"name": "="}]}, {"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "B"}]}]}]}, {"BinaryOperator": [{"name": "-"}]}, {"VarName": [{"Name": [{"String": [{"name": "A"}]}]}]}]}]}]}]}, {"ExpressionStatement": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "i"}]}]}]}, {"BinaryOperator": [{"name": "="}]}, {"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "g"}]}]}]}, {"BinaryOperator": [{"name": "*"}]}, {"WrappedExpression": [{"ParenExpression": [{"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "v"}]}]}]}, {"BinaryOperator": [{"name": "-"}]}, {"VarName": [{"Name": [{"String": [{"name": "e"}]}]}]}]}]}]}]}]}]}]}]}]}]}, {"DerivativeBlock": [{"Name": [{"String": [{"name": "state"}]}]}, {"StatementBlock": [{"ExpressionStatement": [{"DiffEqExpression": [{"BinaryExpression": [{"VarName": [{"PrimeName": [{"String": [{"name": "A"}]}, {"Integer": [{"name": "1"}]}]}]}, {"BinaryOperator": [{"name": "="}]}, {"WrappedExpression": [{"BinaryExpression": [{"UnaryExpression": [{"UnaryOperator": [{"name": "-"}]}, {"VarName": [{"Name": [{"String": [{"name": "A"}]}]}]}]}, {"BinaryOperator": [{"name": "/"}]}, {"VarName": [{"Name": [{"String": [{"name": "tau1"}]}]}]}]}]}]}]}]}, {"ExpressionStatement": [{"DiffEqExpression": [{"BinaryExpression": [{"VarName": [{"PrimeName": [{"String": [{"name": "B"}]}, {"Integer": [{"name": "1"}]}]}]}, {"BinaryOperator": [{"name": "="}]}, {"WrappedExpression": [{"BinaryExpression": [{"UnaryExpression": [{"UnaryOperator": [{"name": "-"}]}, {"VarName": [{"Name": [{"String": [{"name": "B"}]}]}]}]}, {"BinaryOperator": [{"name": "/"}]}, {"VarName": [{"Name": [{"String": [{"name": "tau2"}]}]}]}]}]}]}]}]}]}]}, {"NetReceiveBlock": [{"Argument": [{"Name": [{"String": [{"name": "weight"}]}]}, {"Unit": [{"String": [{"name": "uS"}]}]}]}, {"StatementBlock": [{"ExpressionStatement": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "A"}]}]}]}, {"BinaryOperator": [{"name": "="}]}, {"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "A"}]}]}]}, {"BinaryOperator": [{"name": "+"}]}, {"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "weight"}]}]}]}, {"BinaryOperator": [{"name": "*"}]}, {"VarName": [{"Name": [{"String": [{"name": "factor"}]}]}]}]}]}]}]}]}]}, {"ExpressionStatement": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "B"}]}]}]}, {"BinaryOperator": [{"name": "="}]}, {"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "B"}]}]}]}, {"BinaryOperator": [{"name": "+"}]}, {"WrappedExpression": [{"BinaryExpression": [{"VarName": [{"Name": [{"String": [{"name": "weight"}]}]}]}, {"BinaryOperator": [{"name": "*"}]}, {"VarName": [{"Name": [{"String": [{"name": "factor"}]}]}]}]}]}]}]}]}]}]}]}]}
```

Comment:
Rather pedantic: Since this is a single line, changing a single character will replace the whole line. Makes it harder for Git to function correctly; and harder for us to merge conflicts.

Reply:
I'm not a huge fan either, if we change the output format/API at some point, we'd need to change all of the test data as well...should I just remove the test data for now and only make sure the call to `driver.parse_string(nmodl_string)` succeeds (before it was failing because I erroneously returned the path to the file instead of its contents)?

Reply:
The files are input for the test, but I generated them with NMODL itself, so if we change any part of the API of NMODL, we'd need to regenerate them.

Reply:
Then the, pedantic, suggestion is to:
```
json.dump(f, json.load(f), index=2)
```
To have them formatted in a more human readable manner. This might help Git produce better diffs, which likely only matters if there's ever a conflict; and we want to check something, instead of just brutally regenerating and overwriting.

Reply:
The (possibly minor) problem is that the PR then adds about 8000 extra lines of code, so I am actually not that enthusiastic about including the files in the first place 😅 but if you think it makes sense I can add them. Note that the only functionality these new tests should bring is to try to parse the example mod files, testing the contents themselves of any mod files should probably be done in a separate test.

Reply:
I don't think they're big enough to substantially slow down `git clone`, so it's not as much of an issue. OTOH, if you really don't want them in here, there's a subrepository dedicated to "reference" files, called `nmodl-references` (now I which it were `nmodl-data`).

We can move the location of the to make it work for both usecases.

Reply:
I've removed the actual comparison of the files for now (in the tests we just make sure we can read them), and opened up a separate issue for this https://github.com/BlueBrain/nmodl/issues/1187